docs/DesignDocs/ReproducibleBuildUsingNix.md

```diff
@@ -0,0 +1,397 @@
+  '';
+}
+```
+  
+In the above, let is the initialisation/declaration part of the expression. The actual expression proper begins after
+"in".  The "rec" keyword is used to allow the function definition to be visible to itself.
+
+nix-instantiate will produce the store derivation in the file :
+```
+/nix/store/znmj0axqf4v5mjwswd9bq3dvxf0jcsah-test-environment.drv
+```
+which will contain the nix expression with all attributes assigned with concrete values:
+```
+Derive ( [ ( "out"
+           , "/nix/store/7v249vqann9jrnvb40wri7x4kqr38rp2-test-environment"
+           , ""
+           , "" ) ]
+       , [ ( "/nix/store/029h9shccppyiw1l7qsk6xp0grxgzzbb-stdenv-linux.drv"
+           , ["out"] )
+         , ( "/nix/store/20vwa6qpx8w3ar66x1fmrjlwy86c7b71-bash-4.4-p23.drv"
+           , ["out"] )
+         , ( "/nix/store/nhfvqin361648cnwi3v327jsw9gz48ah-helloWorld.drv"
+           , ["out"] ) ]
+       , [ "/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh"
+         , "/nix/store/l99yzhb3ngxyrfnmglmh98p7f602id2r-xxx" ]
+       , "x86_64-linux"
+       , "/nix/store/hrpvwkjz04s9i4nmli843hyw9z4pwhww-bash-4.4-p23/bin/bash"
+       , [ "-e"
+         , "/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh" ]
+       , [ ( "buildInputs"
+           , "/nix/store/qsgajw2yhrf33qgibsvd9y8zaw64w7j7-helloWorld" )
+         , ( "builder"
+           , "/nix/store/hrpvwkjz04s9i4nmli843hyw9z4pwhww-bash-4.4-p23/bin/bash" )
+         , ("configureFlags", "")
+         , ("depsBuildBuild", "")
+         , ("depsBuildBuildPropagated", "")
+         , ("depsBuildTarget", "")
+         , ("depsBuildTargetPropagated", "")
+         , ("depsHostHost", "")
+         , ("depsHostHostPropagated", "")
+         , ("depsTargetTarget", "")
+         , ("depsTargetTargetPropagated", "")
+         , ("doCheck", "")
+         , ("doInstallCheck", "")
+         , ( "installPhase"
+           , "mkdir $out\ncp -r /nix/store/qsgajw2yhrf33qgibsvd9y8zaw64w7j7-helloWorld $out/\n" )
+         , ("name", "test-environment")
+         , ("nativeBuildInputs", "")
+         , ( "out"
+           , "/nix/store/7v249vqann9jrnvb40wri7x4kqr38rp2-test-environment" )
+         , ("outputs", "out")
+         , ("patches", "")
+         , ("propagatedBuildInputs", "")
+         , ("propagatedNativeBuildInputs", "")
+         , ("src", "/nix/store/l99yzhb3ngxyrfnmglmh98p7f602id2r-xxx")
+         , ( "stdenv"
+           , "/nix/store/sm7kk5n84vaisqvhk1yfsjqls50j8s0m-stdenv-linux" )
+         , ("strictDeps", "")
+         , ("system", "x86_64-linux") ] )
+```
+
+The derivation can then be used to build the output using either
+```
+   nix-store --realize
+```
+or more compactly
+```
+   nix-build test-environment 
+```
+will perform both instantiate and realise. The result is a build output located in the nix-store.
+```
+/nix/store/syibqm72433pzxd46hyw3xbr5swq286j-test-environment
+```
+which contains:
+```
+$ ls /nix/store/syibqm72433pzxd46hyw3xbr5swq286j-test-environment
```

Comment:
Does nix work for Windows builds as well?  The title of this PR, the filename, and the intro in this doc are not Linux specific.  But various places herein only mention Linux.

Reply:
Nix is completely foreign to windows, and I am unaware of any move to port it.  Currently Visual C compiler and linker brand object files with time/date/user/permissions etc information that make it impossible to do a reproducible build.  A repro build could be done using clang, but I suspect that would be unacceptable to most windows developers.

Reply:
I did change the PR title and comment.